### PR TITLE
Add remove

### PR DESCRIPTION
**Pull request by diegoyon**
### To Do List: Add & Remove

- Removed hardcoded items from the task array.
- Implemented the create, edit and remove functions.
- Added functions to update indexes in task array and DOM with any changes.
- Added local storage functionalities.

### CODE REVIEW

File: src/modules/tasks.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    this.list.push(task);
+  }
+
+  removeTask(index) {
+    this.list = this.list.filter((item) => item.index !== index);
+    // update indices
+    for (let i = 0; i < this.list.length; i += 1) {
+      this.list[i].index = i + 1;
+    }
+  }
```
Comment (Mre55) [inline src/modules/tasks.js:21]: Kindly, implement `Clear all completed` button as shown [here](https://www.youtube.com/watch?v=AcUd-_Yjjqg) in the video the user can remove all the completed items. Also, implement the CSS when the user checks the task completed as the video.

**Review by Mre55 — CHANGES_REQUESTED** [review]
**Hi @diegoyon,**

You have done a great job so far! 💪 👏 
You still need to work on a few issues before moving on to the next project, but you're almost there!

## To highlight 🎯

- The Pull Request is following the GitHub Flow ✅ 
- The PR title and description are professional and descriptive ✅ 
- The README is well documented ✅ 
- Linters are working properly ✅ 

## Required Changes ♻️🛠️ 
_Check the comments under the review._





## Optional suggestions 🟡 
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/modules/tasks.js (hunk 1 above)
Comment (diegoyon) [reply src/modules/tasks.js:21]: Those changes are not the requirements of this part of the project. Those are the requirements for tomorrow's part of the project.

Comment (Mre55) [reply src/modules/tasks.js:21]: @diegoyon, I see now it's tomorrow's project. You can submit what you have and the next reviewer will approve it.

File: src/modules/tasks.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+  editTask(newDescription, index) {
+    // console.log(this.list[].description);
```
Comment (gtekle) [inline src/modules/tasks.js:24]: - [ Optional] Kindly remove unused commented out code for a cleaner code base. :+1:

**Review by gtekle — APPROVED** [review]
Hi @diegoyon :wave: , 

# STATUS: APPROVED :tada: :tada: :tada: :100: 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
brilliant-well-done

## Highlights
:heavy_check_mark: Clean and well organized code
:heavy_check_mark: Great implementation of the project requirements
:heavy_check_mark: Nice UI
:heavy_check_mark: Descriptive PR title and summary
:heavy_check_mark: Meaningful commit messages

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**